### PR TITLE
Admin message list

### PR DESCRIPTION
**Pull request by notmd**
- Implement message list page
- Extract common logic to display user's display name into new  `UserDisplayNameCell`component
- Unify `/api/admin/user_messages` endpoint with `/api/admin/messages`
- Allow click to row to expand message content
image

### CODE REVIEW

File: website/src/components/Messages/AdminMessageTable.tsx [hunk 1]
```diff
@@ -42,45 +49,87 @@ const DateDiff = ({ children }: { children: string | Date | number }) => {
...
-export const AdminMessageTable = (props: StrictOmit<DataTableProps<Message>, "columns">) => {
+export const AdminMessageTable = ({ userId, includeUser }: { userId?: string; includeUser?: boolean }) => {
   const [deleteMessageId, setDeleteMessageId] = useState<string | null>(null);
+  const [activeMessageId, setActiveMessageId] = useState<string>();
+  const { pagination, toNextPage, toPreviousPage } = useCursorPagination();
+  const {
+    data: res,
+    isLoading,
+    error,
+    mutate,
```
Comment (AbdBarho) [inline website/src/components/Messages/AdminMessageTable.tsx:60]: would be great to rename `mutate` to something that reflects the actual behavior.

File: website/src/components/Messages/AdminMessageTable.tsx [hunk 2]
```diff
@@ -152,7 +204,14 @@ export const AdminMessageTable = (props: StrictOmit<DataTableProps<Message>, "co
...
 
   const { t } = useTranslation(["common", "message"]);
-
+  const rowProps: DataTableRowPropsCallback<Message> = useCallback((row) => {
+    return {
+      onClick: () => {
+        setActiveMessageId(row.original.id);
+      },
+    };
+  }, []);
```
Comment (AbdBarho) [inline website/src/components/Messages/AdminMessageTable.tsx:213]: `setActiveMessageId` as dependency?

File: website/src/pages/admin/messages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+export default MessageList;
+
+export const getServerSideProps: GetServerSideProps = async ({ locale = "en" }) => {
+  return {
+    props: {
+      ...(await serverSideTranslations(locale, ["common", "labelling", "message"])),
+    },
+  };
+};
```
Comment (AbdBarho) [inline website/src/pages/admin/messages/index.tsx:36]: would this work? https://github.com/LAION-AI/Open-Assistant/blob/3c4bf803b1e50d27cb767593433235ddb033662f/website/src/pages/messages/index.tsx#L8

File: website/src/components/DataTable/useCursorPagination.ts [hunk 4]
```diff
@@ -10,26 +10,26 @@ export interface CursorPaginationState {
...
   const [pagination, setPagination] = useState<CursorPaginationState>({ cursor: "", direction: "forward" });
 
-  const toPreviousPage = (data: undefined | { prev?: string; next?: string }) => {
+  const toPreviousPage = useCallback((data: undefined | { prev?: string; next?: string }) => {
     setPagination({
       cursor: data?.prev || "",
       direction: "back",
     });
-  };
+  }, []);
```
[the remark this answers is not in the document]
Comment (notmd) [reply website/src/components/DataTable/useCursorPagination.ts:23]: Eslint doesn't mark it as an error. Looks like it knows `setPagination` is const

File: website/src/components/DataTable/useCursorPagination.ts [hunk 5]
```diff
@@ -10,26 +10,26 @@ export interface CursorPaginationState {
...
    */
   direction: "forward" | "back";
-}
+};
 
 export const useCursorPagination = () => {
   const [pagination, setPagination] = useState<CursorPaginationState>({ cursor: "", direction: "forward" });
 
-  const toPreviousPage = (data: undefined | { prev?: string; next?: string }) => {
+  const toPreviousPage = useCallback((data: undefined | { prev?: string; next?: string }) => {
```
[the remark this answers is not in the document]
Comment (notmd) [reply website/src/components/DataTable/useCursorPagination.ts:18]: There is a small difference between them. `data?: { prev?: string; next?: string }` means an optional parameter, you can pass or not. `data: undefined | { prev?: string; next?: string }` means you have to explicit pass `undefined` or ` { prev?: string; next?: string }`. You might forget to pass data when using an optional parameters.

File: website/src/components/Messages/AdminMessageTable.tsx [hunk 6]
```diff
@@ -42,45 +49,87 @@ const DateDiff = ({ children }: { children: string | Date | number }) => {
   );
 };
 
-export const AdminMessageTable = (props: StrictOmit<DataTableProps<Message>, "columns">) => {
+export const AdminMessageTable = ({ userId, includeUser }: { userId?: string; includeUser?: boolean }) => {
```
[the remark this answers is not in the document]
Comment (notmd) [reply website/src/components/Messages/AdminMessageTable.tsx:52]: Yes

File: website/src/pages/admin/messages/index.tsx (hunk 3 above)
Comment (notmd) [reply website/src/pages/admin/messages/index.tsx]: Done

**Review by AbdBarho — APPROVED** [review]

**Pull request merged**